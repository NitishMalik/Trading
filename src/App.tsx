import React from 'react';
import './App.css';

import 'bootstrap/dist/css/bootstrap.min.css';

import Template from './common/Template';

import OrderContainer from './components/Order/OrderContainer';

function App() {
  return (
    <div className="App">
      <Template>
        <OrderContainer />
      </Template>
    </div>
  );
}

export default App;
