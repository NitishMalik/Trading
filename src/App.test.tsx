import React from 'react';
import {render} from '@testing-library/react';

import App from './App';

test('generate snapshot for APP', () => {
  const {container} = render(<App />);
  expect(container).toMatchSnapshot();
});
