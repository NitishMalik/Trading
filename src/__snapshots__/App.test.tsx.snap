// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate snapshot for APP 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <span
        class="navbar-brand"
      >
        Stock Trading
      </span>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="navbar-right navbar-nav"
        />
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="order-screen container"
      >
        <form
          class=""
        >
          <div
            class="form-group row"
          >
            <label
              class="form-label col-form-label col-sm-2"
              for="stockName"
            >
              Stock Name
            </label>
            <div
              class="col-sm-10"
            >
              <input
                class="bold form-control-plaintext"
                data-testid="stock-name-ele"
                id="stockName"
                readonly=""
                value="Adani Green"
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="form-label col-form-label col-sm-2"
              for="stockPrice"
            >
              Stock Price
            </label>
            <div
              class="col-sm-5"
            >
              <input
                class="form-control"
                data-testid="stock-price-ele"
                id="stockPrice"
                placeholder="0.0"
                type="text"
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="form-label col-form-label col-sm-2"
              for="formPlaintextPassword"
            >
              Quantity
            </label>
            <div
              class="col-sm-5"
            >
              <input
                class="form-control"
                data-testid="quantity-ele"
                id="formPlaintextPassword"
                placeholder="0"
                type="text"
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="form-label col-form-label col-sm-2"
              for="formPlaintextPassword"
            >
              Transaction Type
            </label>
            <div
              class="no-padding--left col-sm-2"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  data-testid="transaction-buy"
                  id="transaction-type--buy"
                  type="radio"
                />
                <label
                  class="form-check-label"
                  for="transaction-type--buy"
                  title=""
                >
                  Buy
                </label>
              </div>
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  data-testid="transaction-sell"
                  id="transaction-type--sell"
                  type="radio"
                />
                <label
                  class="form-check-label"
                  for="transaction-type--sell"
                  title=""
                >
                  Sell
                </label>
              </div>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="form-label col-form-label col-sm-2"
            />
            <div
              class="no-padding--left col-sm-2"
            >
              <button
                class="btn btn-primary"
                data-testid="place-order-button"
                type="submit"
              >
                Place Order
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <footer>
      All rights Received
    </footer>
  </div>
</div>
`;
